resource/systemd/unit/unit.go

```diff
@@ -0,0 +1,501 @@
+// Copyright © 2016 Asteris, LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package unit
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/asteris-llc/converge/resource"
+	"github.com/asteris-llc/converge/resource/systemd"
+	"github.com/coreos/go-systemd/dbus"
+	"github.com/hashicorp/go-multierror"
+)
+
+// Unit represents a systemd units
+// NOTE: This unit does not parallelize with itself
+type Unit struct {
+	Name          string
+	Active        bool
+	UnitFileState systemd.UnitFileState
+	StartMode     systemd.StartMode
+	Timeout       time.Duration
+
+	resource.TaskStatus
```

Comment:
It's recommended that you embed an actual implementation of the status field, e.g. `*resource.Status`, that will make the fields accessible from `lookup`.